geps/gep-1742.md

```diff
@@ -301,8 +301,134 @@ Could not find any HTTP specific timeouts. PRs welcomed. 😊
 
 ## API
 
-TBD.
+The above diagrams show that there are many different kinds of configurable timeouts
+supported by Gateway implementations: connect, idle, request, upstream, downstream.
+Although there may be opportunity for the specification of a common API for more of
+them in the future, this GEP will focus on the L7 timeouts in HTTPRoutes that are
+most valuable to clients.
+
+From the above analysis, it appears that, at a minimum, implementations are capable of
+supporting the configuration of simple client downstream request timeouts on HTTPRoute
+rules. This is a relatively small addition that would benefit many users.
+
+A client HTTP request to a gateway may result in more than one call to the destination
+backend service, for example, if automatic retries are supported. Some implementations
+support configuring a timeout for the individual backend requests, seperate from the
+overall client request timeout. Adding non-Core support for this would also benefit
+many users.
+
+### GO
+
+```go
+type HTTPRouteRule struct {
+	// Timeouts defines the timeouts that can be configured for an HTTP request.
+	//
+	// Support: Core
+	//
+	// +optional
+	// <gateway:experimental>
+	Timeouts *HTTPRouteTimeouts `json:"timeouts,omitempty"`
+
+	// ...
+}
+
+// HTTPRouteTimeouts defines timeouts that can be configured for an HTTPRoute.
+// Timeout values are formatted like 1h/1m/1s/1ms as parsed by Golang time.ParseDuration
+// and MUST BE >= 1ms.
+type HTTPRouteTimeouts struct {
+	// Request specifies a timeout for responding to client HTTP requests,
+	// disabled by default.
+	//
+	// For example, the following rule will timeout if a client request is taking
+	// longer than 10 seconds to complete:
+	//
+	// ```
+	// rules:
+	// - timeouts:
+	//     request: 10s
+	//   backendRefs:
+	//     ...
+	// ```
```

Comment:
Although we've done it in some places throughout the API, I'd prefer to avoid adding new YAML snippets in our godocs because they're quite painful to support when generating docs.